internal/handler/msg_createuser.go

```diff
@@ -29,6 +37,116 @@ func (h *Handler) MsgCreateUser(ctx context.Context, msg *wire.OpMsg) (*wire.OpM
 		return nil, lazyerrors.Error(err)
 	}
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/1491
-	return nil, common.Unimplemented(document, document.Command())
+	dbName, err := common.GetRequiredParam[string](document, "$db")
+	if err != nil {
+		return nil, err
+	}
+
+	if dbName != "$external" && !document.Has("pwd") {
+		return nil, handlererrors.NewCommandErrorMsg(
+			handlererrors.ErrBadValue,
+			"Must provide a 'pwd' field for all user documents, except those with '$external' as the user's source db",
+		)
+	}
+
+	// https://www.mongodb.com/docs/manual/reference/command/createUser/
+
+	var username string
+	username, err = common.GetRequiredParam[string](document, document.Command())
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err = common.UnimplementedNonDefault(document, "customData", func(v any) bool {
+		if v == nil || v == types.Null {
+			return true
+		}
+
+		cd, ok := v.(*types.Document)
+		return ok && cd.Len() == 0
+	}); err != nil {
+		return nil, err
+	}
+
+	if _, err = common.GetRequiredParam[*types.Array](document, "roles"); err != nil {
+		var ce *handlererrors.CommandError
+		if errors.As(err, &ce) && ce.Code() == handlererrors.ErrBadValue {
+			return nil, handlererrors.NewCommandErrorMsg(
+				handlererrors.ErrMissingField,
+				"BSON field 'createUser.roles' is missing but a required field",
+			)
+		}
+
+		return nil, lazyerrors.Error(err)
+	}
+
+	if err = common.UnimplementedNonDefault(document, "roles", func(v any) bool {
+		if v == nil || v == types.Null {
```

Comment:
I think this line might conflict with the previous check where we ask `roles` to be required (so `v` can't be `nil` here).